tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py

```diff
@@ -180,8 +179,94 @@ def read(self, data_id):
 class ScipyImageReader(BaseReader):
     __provider__ = 'scipy_imread'
 
+    @staticmethod
+    def _from_image(image):
+        if image.mode == 'P':
+            image = image.convert('RGBA') if 'transparency' in image.info else image.convert('RGB')
+
+        return np.array(image)
+
+    @staticmethod
+    def _process_2d(data, shape):
+        shape = (shape[1], shape[0])  # columns show up first
+        bytedata = ScipyImageReader._bytescale(data)
+        image = Image.frombytes('L', shape, bytedata.tostring())
+
+        return image
+
+    @staticmethod
+    def _process_3d(data, shape):
+        # if here then 3-d array with a 3 or a 4 in the shape length.
+        # Check for 3 in datacube shape --- 'RGB' or 'YCbCr'
+        ca = np.flatnonzero(np.asarray(shape) == 3) if 3 in shape else np.flatnonzero(np.asarray(shape) == 4)
+        if not np.size(ca):
+            raise ValueError("Could not find channel dimension.")
+        ca = ca[0]
```

Comment:
This channel dimension searching logic seems really dodgy to me. Surely in our case the channel dimension is always the same?